scripts/ci-entrypoint.sh

```diff
@@ -79,28 +83,31 @@ select_cluster_template() {
         # shellcheck source=scripts/ci-build-kubernetes.sh
         source "${REPO_ROOT}/scripts/ci-build-kubernetes.sh"
         export CLUSTER_TEMPLATE="test/dev/cluster-template-custom-builds.yaml"
-    elif [[ -n "${CI_VERSION:-}" ]] || [[ -n "${USE_CI_ARTIFACTS:-}" ]] || [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
+    elif [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]] || [[ -n "${CI_VERSION:-}" ]]; then
         # export cluster template which contains the manifests needed for creating the Azure cluster to run the tests
-        GOPATH="$(go env GOPATH)"
-        if ls "${GOPATH}"/src/k8s.io/kubernetes; then
-            KUBERNETES_BRANCH="$(cd "${GOPATH}/src/k8s.io/kubernetes" && git rev-parse --abbrev-ref HEAD)"
-        fi
-        if [[ "${KUBERNETES_BRANCH:-}" =~ "release-" ]]; then
-            CI_VERSION_URL="https://dl.k8s.io/ci/latest-${KUBERNETES_BRANCH/release-}.txt"
-        elif [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
-            CI_VERSION_URL="https://dl.k8s.io/ci/${KUBERNETES_VERSION}.txt"
-        else
-            CI_VERSION_URL="https://dl.k8s.io/ci/latest.txt"
-        fi
         export CLUSTER_TEMPLATE="test/ci/cluster-template-prow-ci-version.yaml"
-        export CI_VERSION="${CI_VERSION:-$(curl -sSL ${CI_VERSION_URL})}"
-        export KUBERNETES_VERSION="${CI_VERSION}"
+        if [[ -z "${CI_VERSION:-}" ]]; then
+            if [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
+                CI_VERSION_URL="https://dl.k8s.io/ci/${KUBERNETES_VERSION}.txt"
+            else
+                CI_VERSION_URL="https://dl.k8s.io/ci/latest.txt"
```

Comment:
don't think we need this else here, `CI_VERSION_URL="https://dl.k8s.io/ci/${KUBERNETES_VERSION}.txt"` is equivalent to `CI_VERSION_URL="https://dl.k8s.io/ci/latest.txt"` when `KUBERNETES_VERSION is `latest` and there is no more defaulting to "latest" when `USE_CI_ARTIFACTS` is true.

This whole block (line 89-95) could just be 

```
if [[ "${KUBERNETES_VERSION:-}" =~ "latest" ]]; then
                CI_VERSION_URL="https://dl.k8s.io/ci/${KUBERNETES_VERSION}.txt"
fi
```

Reply:
good catch!